src/components/Files/FileUploadDialog.tsx

```diff
@@ -40,14 +41,26 @@ export default function FileUploadDialog({
     onOpenChange(open);
   };
 
+  const [previewUrls, setPreviewUrls] = useState<string[]>([]);
+
+  useEffect(() => {
+    const urls = fileUpload.files
+      .filter((file) => file.type.startsWith("image/"))
+      .map((file) => URL.createObjectURL(file));
+    setPreviewUrls(urls);
+    return () => {
+      urls.forEach((url) => URL.revokeObjectURL(url));
+    };
+  }, [fileUpload.files]);
+
   return (
     <Dialog
       open={open}
       onOpenChange={handleDialogClose}
       aria-labelledby="file-upload-dialog"
     >
       <DialogContent
-        className="mb-8 rounded-lg p-5 max-w-fit md:max-w-[30rem]"
+        className="mb-8 rounded-lg p-5 max-w-fit md:max-w-[30rem] shadow-[0_3px_10px_rgba(0,0,0,0.12)] hover:shadow-[0_4px_14px_rgba(0,0,0,0.16)] transition-all duration-200"
```

Comment:
let's not do custom ones like this unless absolutely necessary, make sure of built-in tailwind classes.